services/snapshotter/client.go

```diff
@@ -22,6 +26,138 @@ func NewClient(host string) *Client {
 	return &Client{host: host}
 }
 
+// takes a request object, writes a Base64 encoding to the tcp connection, and then sends the request to the snapshotter service.
+// returns a mapping of the uploaded metadata shardID's to actual shardID's on the destination system.
+func (c *Client) UpdateMeta(req *Request, upStream io.Reader) (map[uint64]uint64, error) {
+	var err error
+
+	var b bytes.Buffer
+
+	// Connect to snapshotter service.
+	conn, err := tcp.Dial("tcp", c.host, MuxHeader)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+
+	if _, err := conn.Write([]byte{byte(req.Type)}); err != nil {
+		return nil, err
+	}
+
+	if err := json.NewEncoder(conn).Encode(req); err != nil {
+		return nil, fmt.Errorf("encode snapshot request: %s", err)
+	}
+
+	if n, err := io.Copy(conn, upStream); (err != nil && err != io.EOF) || n != req.UploadSize {
+		return nil, fmt.Errorf("error uploading file: err=%v, n=%d, uploadSize: %d", err, n, req.UploadSize)
+	}
+
+	if n, err := b.ReadFrom(conn); err != nil || n == 0 {
+		return nil, fmt.Errorf("updating metadata on influxd service failed: err=%v, n=%d", err, n)
+	}
+
+	resp := b.Bytes()
```

Comment:
In this scenario, prefer `ioutil.ReadAll` rather than creating a `bytes.Buffer`, using `ReadFrom`, and then only calling `Bytes()`.

Reply:
👍